sklearn/metrics/_scorer.py

```diff
@@ -322,8 +315,6 @@ def _score(self, method_caller, clf, X, y, sample_weight=None, **kwargs):
                     y_pred = np.vstack([p[:, -1] for p in y_pred]).T
 
         scoring_kwargs = copy.deepcopy(self._kwargs)
-        if sample_weight is not None:
```

Comment:
I wonder whether there are cases where this could break backwards compatibility, where the metric now gets passed sample_weight=None